linux-firmware: allow cherry-picks at build time

graysky2 opened the pull request:
This commit add the flexibility of cherry-picking post-release commits such as backports and reverts at build time by switching to a git  source and by adding an optional build step override.

This change needed when post-release changes are made to binary files since quilt does not support binary diffs but git does.

Additionally, this commit incorporates two post-20250613 version back-ports to fix a breaking bug affecting users with some AMD GPUs.

References to the two commits we are cherry-picking:
1. https://gitlab.archlinux.org/archlinux/packaging/packages/linux-firmware/-/issues/12
2. https://gitlab.archlinux.org/archlinux/packaging/packages/linux-firmware/-/issues/16

DragonBluep commented on package/firmware/linux-firmware/amd.mk:
```diff
@@ -14,6 +14,9 @@ define Package/amdgpu-firmware/install
 		$(1)/lib/firmware/amdgpu
+
+	$(INSTALL_DATA) ./files/dcn_3_1_5_dmcub.bin $(1)/lib/firmware/amdgpu
+	$(INSTALL_DATA) ./files/dcn_4_0_1_dmcub.bin $(1)/lib/firmware/amdgpu
```
This is not a good idea to include binary files. This will significantly increase the repository size. We can directly download these firmware from Git. An example:
https://github.com/openwrt/openwrt/blob/764576eb5b6bae6c7142bd9e560f8c68da177b0d/package/boot/uboot-mediatek/Makefile#L14

https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/plain/amdgpu/dcn_3_1_5_dmcub.bin?id=9096bad65cb92da2925fe9ad891cbd4961fa6cda
https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/amdgpu/dcn_4_0_1_dmcub.bin?id=9096bad65cb92da2925fe9ad891cbd4961fa6cda

graysky2 replied:
Please see latest commit

graysky2 commented:
CI reports:
```
Some package Makefiles requires fix. (run 'make package/check FIXUP=1' and force push this pr)
You can also check the provided artifacts with the refreshed patch from this CI run.
```

But running the above ends in error code 0 yet no changes have been made.
```
make package/linux-firmware/check FIXUP=1
```

graysky2 committed:
linux-firmware: allow cherry-picks at build time

This commit add the flexibility of cherry-picking post-release commits
such as backports and reverts at build time by switching to a git
source and by adding an optional build step override.

This change needed when post-release changes are made to binary files since
quilt does not support binary diffs but git does.

Additionally, this commit incorporates two post-20250613 version back-ports
to fix a breaking bug affecting users with some AMD GPUs.

References to the two commits we are cherry-picking:
1. https://gitlab.archlinux.org/archlinux/packaging/packages/linux-firmware/-/issues/12
2. https://gitlab.archlinux.org/archlinux/packaging/packages/linux-firmware/-/issues/16

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
@DragonBluep - I change this PR avoiding the mods you suggested and instead using git to cherry pick. What do you think?

graysky2 commented:
@nbd168 - since you are hard-coded in the Makefile, please review
